packages/jest-cli/src/search_source.js

```diff
@@ -140,23 +142,42 @@ export default class SearchSource {
     return this._getAllTestPaths(testPathPattern);
   }
 
-  findRelatedTests(allPaths: Set<Path>): SearchResult {
+  findRelatedTests(
+    allPaths: Set<Path>,
+    collectCoverage: boolean,
+  ): SearchResult {
     const dependencyResolver = new DependencyResolver(
       this._context.resolver,
       this._context.hasteFS,
     );
-    return {
-      tests: toTests(
-        this._context,
-        dependencyResolver.resolveInverse(
-          allPaths,
-          this.isTestFilePath.bind(this),
-          {
-            skipNodeResolution: this._context.config.skipNodeResolution,
-          },
-        ),
+
+    const tests = toTests(
+      this._context,
+      dependencyResolver.resolveInverse(
+        allPaths,
+        this.isTestFilePath.bind(this),
+        {
+          skipNodeResolution: this._context.config.skipNodeResolution,
+        },
       ),
-    };
+    );
+
+    // If we are collecting coverage, also return collectCoverageFrom patterns
+    if (collectCoverage) {
+      const collectCoverageFrom = Array.from(allPaths).map(filename => {
+        filename = _replaceRootDirInPath(
+          this._context.config.rootDir,
+          filename,
+        );
+        return path.isAbsolute(filename)
+          ? path.relative(this._context.config.rootDir, filename)
+          : filename;
+      });
+
+      return {collectCoverageFrom, tests};
```

Comment:
We're going polymorphic here, let's just set `collectCoverageFrom` to undefined by default

Reply:
Or empty array, whatever is easier

Reply:
Yeah sure 👍